src/interp.rs

```diff
@@ -145,15 +150,46 @@ impl<'a, 'tcx> Machine<'a, 'tcx> {
                 self.eval_basic_block(target)
             }
             TerminatorKind::Return =>  {
-                // Move the ret_addr, otherwise we borrow twice.
-                let ret_addr = self.cur_frame().ret_addr;
-                match ret_addr {
+                // Move the ret_block, otherwise we borrow twice.
+                let ret_block = self.cur_frame().ret_block;
+                self.pop_frame();
+                match ret_block {
                     Some(bb) => {
                         self.eval_basic_block(bb)
                     },
                     None => exit(0),
                 }
             },
+            TerminatorKind::Call {
+                ref func,
+                ref args,
+                ref destination,
+                ref cleanup
+            } => {
+                let (ret_val, ret_block) = match *destination {
+                    Some((ref place, bb)) => (Some(self.eval_place(place)), Some(bb)),
+                    None => (None, None)
+                };
+                let func = self.eval_operand(func);
+                let (fn_def, sig) = match func.ty.sty {
+                    ty::TyFnDef(def_id, substs) => {
+                        let instance = ty::Instance::resolve(
+                            self.tcx,
+                            self.tcx.param_env(def_id),
+                            def_id,
+                            substs,
+                        ).unwrap();
+                        let sig = func.ty.fn_sig(self.tcx);
+                        (instance, sig)
```

Comment:
Is `sig` used?

Reply:
Nope, good spot, it's leftover code from a previous approach to implementation!